### PR TITLE
Feature/calibrate inherent rio orientation

### PR DESCRIPTION
**Pull request by JohnGilb**

### CODE REVIEW

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java [hunk 1]
```diff
@@ -103,6 +106,8 @@ public PoseSubsystem(WPIFactory factory, XPropertyManager propManager) {
         totalDistanceY = propManager.createEphemeralProperty("TotalDistanceY", 0.0);
         
         rioRotated = propManager.createPersistentProperty("RioRotated", false);
+        inherentRioPitch = propManager.createPersistentProperty("InherentRioPitch", 0.0);
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: this names ok, but I'll offer 'inherentRioPitchOffset' as an alternative

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java [hunk 2]
```diff
@@ -251,21 +256,32 @@ public PoseResult getDistanceFromLeftRangerfinderToLeftWall() {
     }
     
     public double getRobotPitch() {
-        if (rioRotated.get())
-        {
+        return getRealPitch() - inherentRioPitch.get();
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: and we might eventually want to just roll this into the XGyro?

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java [hunk 3]
```diff
@@ -251,21 +256,32 @@ public PoseResult getDistanceFromLeftRangerfinderToLeftWall() {
...
+    }
+    
+    public double getRobotRoll() {
+        return getRealRoll() - inherentRioRoll.get();
+    }
+    
+    private double getRealPitch() {
```
Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: maybe getRaw?

**Comment by aschokking** [comment]
lgtm;

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java (hunk 2 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: Yeah, definitely sounds like a good idea. I'm gonna add it to the backlog.

File: CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java (hunk 3 above)
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/drive/PoseSubsystem.java]: Raw has the connotation of just returning what the sensor has, whereas this still has the "is rio rotated" aspect. I'm not super happy with the name either, though...

**Pull request merged**